### PR TITLE
Feature/sugrinov add applicant to token store

### PR DESCRIPTION
**Pull request by sugrinov**

### CODE REVIEW

File: web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java [hunk 1]
```diff
@@ -102,4 +115,23 @@
     testService.submit(submitTestRequest.getTestId(), convertedAnswers);
     return new ResponseEntity<>(HttpStatus.OK);
   }
+
+  private ResponseEntity<TestDto> checkUserHasTestAccess(HttpServletRequest httpRequest, Test test) {
+    Optional<String> optionalToken = Optional.ofNullable(httpRequest.getHeader(APPLICANT_TOKEN_KEY));
+
+    boolean tokenNotPresent = !(optionalToken.isPresent() && tokenStore.containsKey(optionalToken.get()));
+    if (tokenNotPresent) {
+      return new ResponseEntity<>(HttpStatus.BAD_REQUEST); // 400
```
Comment (luinnarn) [inline web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java]: Just a suggestion, one possible situation is that a token is present, but not contained in tokenStore, so perhaps the boolean should be named something like `validTokenNotPresent` or `validTokenNotFound`?

Comment (sugrinov) [reply web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java]: Yes that is a valid suggestion, I will change the name. But both of those imply(to me atleast) that token is present in header but it is not present in token store and it doesn't have to be present in the header therefore we have the check. What do you think about `isTokenPresentAndValid`? Maybe it is an overkill but makes most sense to me :laughing:

Comment (luinnarn) [reply web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java]: `isTokenPresentAndValid` is OK but then you would have to negate it, `if (!isTokenPresentAndValid)`, `if (validTokenNotPresent)` simply seems more readable.

File: web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java [hunk 2]
```diff
@@ -41,6 +48,8 @@
   private FormulaService formulaService;
   @Autowired
   private UserService userService;
+  @Autowired
```
Comment (dujfalusi) [inline web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java]: Please fix formatting

File: web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java [hunk 3]
```diff
@@ -102,4 +115,23 @@
     testService.submit(submitTestRequest.getTestId(), convertedAnswers);
     return new ResponseEntity<>(HttpStatus.OK);
   }
+
+  private ResponseEntity<TestDto> checkUserHasTestAccess(HttpServletRequest httpRequest, Test test) {
```
Comment (dujfalusi) [inline web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java]: Shouldn't this, be a part of some user service?

File: web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java (hunk 2 above)
Comment (sugrinov) [reply web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java]: What should be formatted? You mean spaces between services?

File: web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java (hunk 3 above)
Comment (sugrinov) [reply web/src/main/java/eu/execom/testgenerator/web/controller/TestController.java]: Most of the logic here is to check which `Response` should be returned so it makes more sense to be here.

File: web/src/test/java/eu/execom/testgenerator/web/controller/TestControllerTest.java [hunk 4]
```diff
@@ -92,17 +96,24 @@ public void shouldCreateNewTest() throws Exception {
     Test test = new Test();
     test.setId(5L);
 
+    String token = "token";
+
+
```
Comment (dujfalusi) [inline web/src/test/java/eu/execom/testgenerator/web/controller/TestControllerTest.java]: Please remove this unnecessary empty line.

**Pull request merged**